src/gui/DetailsWidget.cpp

```diff
@@ -309,9 +309,17 @@ void DetailsWidget::setDatabaseMode(DatabaseWidget::Mode mode)
     }
     if (mode == DatabaseWidget::ViewMode) {
         if (m_ui->stackedWidget->currentIndex() == GroupPreview) {
-            getSelectedGroup(m_currentGroup);
+            if (m_currentGroup) {
+                getSelectedGroup(m_currentGroup);
+            } else {
+                hideDetails();
+            }
         } else {
-            getSelectedEntry(m_currentEntry);
+            if (m_currentEntry) {
+                getSelectedEntry(m_currentEntry);
```

Comment:
@TheZ3ro shouldn't this logic go in `getSelectedEntry` (and `getSelectedGroup`) instead?

Reply:
Mmmmh yeah, why not

Reply:
@TheZ3ro why functions start with "get*", why not with "set*" or "update*"?